### PR TITLE
[DO_NOT_MERGE] test applications via a route, not a direct endpoint

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by danwinship** [comment]
looks right I think but I don't really know routes

File: test/extended/util/framework.go [hunk 1]
```diff
@@ -1247,6 +1247,24 @@ func GetEndpointAddress(oc *CLI, name string) (string, error) {
 	return fmt.Sprintf("%s:%d", endpoint.Subsets[0].Addresses[0].IP, endpoint.Subsets[0].Ports[0].Port), nil
 }
 
+// GetServiceAddress will return an "ip:port" string for the endpoint.
+func GetServiceAddress(oc *CLI, name string) (string, error) {
```
Comment (ironcladlou) [inline test/extended/util/framework.go]: This was added but I don't see any references to it.

**Comment by ironcladlou** [comment]
Does what it says on the tin, can't speak as to the underlying intent (e.g. was there a desire to make it more of a black box test that exercises routing? Was relying solely on cluster networking considered too shallow a test?)

**Comment by bparees** [comment]
> Does what it says on the tin, can't speak as to the underlying intent (e.g. was there a desire to make it more of a black box test that exercises routing? Was relying solely on cluster networking considered too shallow a test?)

the existing tests don't work in multinode clusters because they relied on direct local access. i'm trying to rework them so they do.  I don't really need @openshift/sig-networking's input on this, that was just prow being aggressive in soliciting reviews.

**Pull request closed without merging**